### PR TITLE
Require approval to run actions for fork pull request

### PR DESCRIPTION
**Pull request by wolfogre**
Currently, Gitea will run actions automatically which are triggered by fork pull request. It's a security risk, people can create a PR and modify the workflow yamls to execute a malicious script.

So we should require approval for first-time contributors, which is the default strategy of a public repo on GitHub, see [Approving workflow runs from public forks](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks).

Current strategy:

- don't need approval if it's not a fork PR;
- always need approval if the user is restricted;
- don't need approval if the user can write;
- don't need approval if the user has been approved before;
- otherwise, need approval.

https://user-images.githubusercontent.com/9418365/217207121-badf50a8-826c-4425-bef1-d82d1979bc81.mov

GitHub has an option for that, you can see that at `/<owner>/<repo>/settings/actions`, and we can support that later.

image

### CODE REVIEW

File: models/migrations/migrations.go [hunk 1]
```diff
@@ -455,6 +455,8 @@ var migrations = []Migration{
 	NewMigration("Add scope for access_token", v1_19.AddScopeForAccessTokens),
 	// v240 -> v241
 	NewMigration("Add actions tables", v1_19.AddActionsTables),
+	// v241 -> v242
+	NewMigration("Add actions tables", v1_19.AddNeedApprovalForActionRun),
```
Comment (lunny) [inline models/migrations/migrations.go]: The description should be changed.

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
@yardenshoham This PR has one migration so that it cannot be backport to old version.

File: models/migrations/migrations.go [hunk 2]
```diff
@@ -461,8 +461,10 @@ var migrations = []Migration{
 	NewMigration("Alter gpg_key_import content TEXT field to MEDIUMTEXT", v1_19.AlterPublicGPGKeyImportContentFieldToMediumText),
 	// v243 -> v244
 	NewMigration("Add exclusive label", v1_19.AddExclusiveLabel),
+	// v244 -> v245
+	NewMigration("Add NeedApproval to actions tables", v1_19.AddNeedApprovalToActionRun),
 
-	// Gitea 1.19.0 ends at v244
+	// Gitea 1.19.0 ends at v245
```
Comment (yardenshoham) [inline models/migrations/migrations.go]: It'll go to 1.20

**Comment by delvh** [comment]
On the other hand, what we could also do is merge #23078, and then backport it.

**Comment by lunny** [comment]
> On the other hand, what we could also do is merge #23078, and then backport it.

That PR is not ready, we should be careful and need more discuss on that side.

**Review by Zettat123 — APPROVED** [review]

**Pull request merged**